duckduckgo/duckduckgo-privacy-extension
# Remove GPC declarativeNetRequest rule when GPC feature is disabled

When the user selects to disable Global Privacy Control (GPC) in the
options page, we must take care to remove the corresponding
declarativeNetRequest rule. Otherwise, the GPC headers will be added
to requests against the user's wishes.

**Reviewer:** @sammacbeth 
**CC:** @jdorweiler 

## Steps to test this PR:
1. Navigate to the GPC test page with the MV3 build https://privacy-test-pages.site/privacy-protections/gpc/
2. Run the tests and save the results
3. Turn off the GPC feature in the options page
4. Run the tests again and save the results
5. Remove the MV3 extension and install the MV2 extension
6. Repeat the tests with/without GPC enabled and compare the results to the ones saved earlier. Make sure they match exactly.

## Automated tests:
